File: backend/Snagged.backend/Snagged.Domain/Entities/CartItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Snagged.Domain.Entities
{
    public class CartItem
    {
        public int Id { get; set; }
        public int CartId { get; set; }
        public Cart Cart { get; set; }
        public int ItemId { get; set; }
        public Item Item { get; set; }
        public int Quantity { get; set; } = 1;
        public DateTime AddedAt { get; set; } = DateTime.Now;
       

    }
}

File: frontend/src/app/shared/models/cart-item.ts
import { Cart } from './cart';
import { ItemModel } from './item.model';

export interface CartItem {
  id: number;
  cartId: number;
  cart?: Cart;
  itemId: number;
  item?: ItemModel;
  quantity: number;
  addedAt: Date | string;
  price: number;
  imageUrl: string;
}
